packages/core/src/components/Logo/types.ts

```diff
@@ -0,0 +1,5 @@
+export type LogoProps = {
+    showName?: boolean;
+    logo?: React.FC;
```

Comment:
One can achieve this by wrapping the component in an arrow function - `() => <SomeLogo size="S" />`.

I do understand though that this is slightly more cumbersome. I followed the approach taken in the `MedlySidenavHeader` so that this is easily compatible with it. It expects the `companyName` and `companyHeader` to be `React.FC<SVG>`. We could update that component to pass the rendered name and logo like 
`<Logo showName={isHovered} name={<companyName />} logo={<companyLogo />} />` to keep it backward compatible.

I personally think that it is fine either way, let me know what you think.